### PR TITLE
SCRP-1346: Update Gradle repo credential from CES_GITHUB_PACKAGES_READ_*

### PR DESCRIPTION
**Pull request by jsoetanto-ces**
Example run from [ms-pos-refund-service](https://github.com/coexservices/ms-pos-refund-service/actions)

Screenshot 2023-03-07 at 2 13 47 pm

**ms-pos-agreement-service**

pos-agreement validate-pr
pos-agreement build

**pos-api**

pos-api validate-pr
pos-api build
pos-api changes

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -50,16 +50,17 @@ jobs:
 
     - name: Project Step Calculator
       run: |
+        echo "MVN_REPO_USERNAME=$( [ ! -z "${{ secrets.github_repo_actor }}" ] && echo ${{ secrets.github_repo_actor }} || echo ${{ secrets.CES_GITHUB_PACKAGES_READ_ACTOR }} )" >> $GITHUB_ENV
```
Comment (kenzhang-ces) [inline .github/workflows/build.yml:53]: Do you really need " >> $GITHUB_ENV"  here?

File: .github/workflows/validate-pr.yml [hunk 2]
```diff
@@ -137,9 +137,10 @@ jobs:
 
       - name: Project Step Calculator
         run: |
+          echo "MVN_REPO_USERNAME=$( [ ! -z "${{ secrets.github_repo_actor }}" ] && echo ${{ secrets.github_repo_actor }} || echo ${{ secrets.CES_GITHUB_PACKAGES_READ_ACTOR }} )" >> $GITHUB_ENV
```
Comment (kenzhang-ces) [inline .github/workflows/validate-pr.yml:140]: secrets.github_token and secrets.github_actor ?

These are what I used in pos-commons and ces-starters 

https://github.com/coexservices/ces-starters/blob/43f75b479004ff81e9581a8112f22463afb211bc/.github/workflows/build-artifact-dev.yml#L33

https://github.com/coexservices/pos-commons/blob/7ec4de3db96d10af183ea0c950853c7b54776c8a/.github/workflows/build-artifact-dev.yml#L32

Comment (jsoetanto-ces) [reply .github/workflows/validate-pr.yml:140]: This workflow has pre-defined arguments. This is handling to read either from this arguments / using the shared organisation username and token

File: .github/workflows/validate-pr.yml [hunk 3]
```diff
@@ -171,8 +172,8 @@ jobs:
         run: |
           mkdir -p $HOME/.gradle
           echo "allprojects { repositories { mavenCentral(); maven { name \"GitHubPackages\"; url uri(\"https://maven.pkg.github.com/coexservices/*\"); credentials { username githubRepoActor; password githubRepoToken } } } }" > ~/.gradle/init.gradle
-          echo "githubRepoActor=${{ secrets.GITHUB_ACTOR }}" > ~/.gradle/gradle.properties
-          echo "githubRepoToken=${{ secrets.GITHUB_TOKEN }}" >> ~/.gradle/gradle.properties
```
Comment (kenzhang-ces) [inline .github/workflows/validate-pr.yml:175]: gosh... secrets.GITHUB_ACTOR / secrets.GITHUB_TOKEN used here...

File: .github/workflows/validate-pr.yml (hunk 2 above)
Comment (kenzhang-ces) [reply .github/workflows/validate-pr.yml:140]: I can't find github_repo_actor and github_repo_token in other project... Not sure how this was defined

Comment (jsoetanto-ces) [reply .github/workflows/validate-pr.yml:140]: https://github.com/coexservices/workflows/blob/733221ef57559d51fe54f1d6ed3af3eef5d7a796/.github/workflows/validate-pr.yml#L24-L27

**Review by aconstanti-ces — APPROVED** [review]

**Pull request merged**